infor-design/enterprise
# 4820 - Lookup/Timepicker/Datepicker accessibility improvements

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR makes some changes to some trigger field components (Lookup, Datepicker, Timepicker) to convert the "trigger" from just an icon to a button, along with adding various missing ARIA roles, descriptive text, and other minor improvements to improve detection of these components by screen readers.

To test this issue, please pre-install Chrome Vox or another screen reader application.

**Related github/jira issue (required)**:
Closes #4820 

**Steps necessary to review your pull request (required)**:

_Test Lookup_:
- Open http://localhost:4000/components/lookup/example-index?layout=nofrills
- Enable Chrome Vox (or other screen reader)
- Focus the first input field.  The screen reader should speak "Products Lookup.  Press Down Arrow to Select. Edit Text, Required"
- Press the tab key.  The Lookup icon is now focusable and should appear focused.  The screen reader should speak "Lookup Trigger Button"
- Test RTL and check for icon/button alignment: http://localhost:4000/components/lookup/example-index?layout=nofrills&locale=ar-SA
- Test classic theme:  http://localhost:4000/components/lookup/example-index?theme=classic&mode=light&layout=nofrills
- Test non-tabbable example - the new second tabstop should be disabled, and the keyboard behavior should work like before this change: http://localhost:4000/components/lookup/example-not-tabbable?theme=new&mode=light&layout=nofrills

_Test Datepicker_:
- Open http://localhost:4000/components/datepicker/example-index?theme=new&mode=light&layout=nofrills
- Enable Chrome Vox (or other screen reader)
- Focus the first input field:.  The screen reader should speak "Date Field. Press Down Arrow to Select. Combo Box Collapsed".
- Press the Tab key.  The trigger icon is now focusable and should appear focused.  The screen reader should speak "Date Picker trigger button"
- Test RTL and check for icon/button alignment: http://localhost:4000/components/datepicker/example-index?theme=new&mode=light&layout=nofrills&locale=ar-SA
- Test classic theme: http://localhost:4000/components/datepicker/example-index?theme=classic&mode=light&layout=nofrills
- Test non-tabbable example - the new second tabstop should be disabled, and the keyboard behavior should work like before this change: http://localhost:4000/components/datepicker/example-not-tabbable?theme=new&mode=light&layout=nofrills
- Test Calendar Toolbar: http://localhost:4000/components/calendar-toolbar/example-index.html
- 
	
_Test Timepicker_:
- Open http://localhost:4000/components/timepicker/example-index?theme=new&mode=light&layout=nofrills
- Enable Chrome Vox
- Focus the first input field.  The screen reader should speak "Timepicker. Use Arrow Keys to Select. Combobox Collapsed."
- Press the Tab Key.  The trigger icon is now focusable and should appear focused. The screen reader should speak "Time picker trigger button"
- Test RTL and check for icon/button alignment: http://localhost:4000/components/timepicker/example-index?theme=new&mode=light&layout=nofrills&locale=ar-SA
- Test classic theme: http://localhost:4000/components/timepicker/example-index?theme=classic&mode=light&layout=nofrills
- Test non-tabbable example - the new second tabstop should be disabled, and the keyboard behavior should work like before this change:  http://localhost:4000/components/timepicker/example-not-tabbable?theme=new&mode=light&layout=nofrills

_Test Datagrid Filter Row:_
- Open https://localhost:4000/components/datagrid/example-filter.html
- Check the Datepicker, Timepicker, and Lookup fields and ensure the icon placement looks right
- Check RTL: https://localhost:4000/components/datagrid/example-filter.html?locale=ar-EG
- Check Classic Theme: http://localhost:4000/components/datagrid/example-filter.html?theme=classic&mode=light

_Test Short Fields_:
- Open http://localhost:4000/components/form/example-compact-mode.html
- Check button/icon alignments on Lookup, Datepicker, and Timepicker. Ensure focus state appears to be within the boundaries of the field.
- Check RTL: http://localhost:4000/components/form/example-compact-mode.html?locale=ar-SA
- Check Classic Theme: http://localhost:4000/components/form/example-compact-mode.html?theme=classic&mode=light&locale=ar-SA

Test any other examples of these components you think may be missing from the list above (I'm sure I missed some)

**Included in this Pull Request**:
